### PR TITLE
DT-1418 Recall a prisoner API

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 1]
```diff
@@ -121,6 +122,24 @@ public String releasePrisoner(
         return offenderNo;
     }
 
+    @ApiResponses({
+        @ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse.class),
+        @ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse.class),
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:127]: Maybe a 403 too?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ProfileType.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Column(name = "PROFILE_CATEGORY")
+    private String category;
+
+    @Column(name = "DESCRIPTION")
+    private String description;
+
+    @Column(name = "MANDATORY_FLAG", nullable = false)
+    @Enumerated(EnumType.STRING)
+    private ActiveFlag mandatory = ActiveFlag.Y;
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ProfileType.java:37]: Does this one need @Default too?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ProfileType.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Column(name = "ACTIVE_FLAG", nullable = false)
+    @Enumerated(EnumType.STRING)
+    @Default
+    private ActiveFlag activeFlag = ActiveFlag.Y;
+
+    @Column(name = "EXPIRY_DATE")
+    private LocalDate endDate;
+
+    @Column(name = "LIST_SEQ", nullable = false)
+    @Builder.Default
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/ProfileType.java:56]: Already importing Default.

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java (hunk 1 above)
Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:127]: yes

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**